### ISSUE TITLE
Disallow class calls from arrays

### ISSUE THREAD

**Issue by gmponos** [squizlabs/PHP_CodeSniffer#1198]
Let's say that I have the following code:

``` php
$myarray['class1'] = new MyClass();
$myarray['class2'] = new MyOtherClass();
```

I need to disallow someone from calling a function like this:

``` php
$result = $myarray['class1']->myFunction();
```

Is there any kind of sniff to do so?

**Comment by aik099** [comment]
1. The questions can be asked in Gitter chat.
2. Not sure there is, because it looks very specific.

To generalize you want to disallow all constructs looking like:

```php
$any_variable_name['any_string_key']->any_method(any_parameters);
```

Looks doable.

**Comment by gmponos** [comment]
Yes.. That's the general idea.. most of the times the classes that are contained in the array are not Instances of the same Class and this messes autocomplete in many cases. Disallowing this kind of call the user will be enforced to assign the Class to a variable first, like:

``` php
/** AnyClass $anyClass */
$anyClass = $any_variable_name['any_string_key'];
$anyClass->any_method(any_parameters);
```

When you are saying Looks doable are you willing to work on it?

**Comment by gsherwood** [comment]
There is no current included sniff to do this. I personally don't have time to work on it

**Comment by aik099** [comment]
> When you are saying Looks doable are you willing to work on it?

Sure. Contact me directly (e.g. over email) if interested.
